src/monitors/radiationFlux.cpp

```diff
@@ -0,0 +1,181 @@
+    PetscFunctionBeginUser;
+
+    // If this is the first output, store a copy of the fluxDm
+    if (sequenceNumber == 0) {
+        DMView(fluxDm, viewer) >> utilities::PetscUtilities::checkError;
+    }
+
+    // Set the output sequence number to the face dm
+    DMSetOutputSequenceNumber(fluxDm, sequenceNumber, time) >> utilities::PetscUtilities::checkError;
+
+    /** Get read access to the local solution information
+     */
+    const PetscScalar* locXArray;
+    VecGetArrayRead(GetSolver()->GetSubDomain().GetSolutionVector(), &locXArray) >> utilities::PetscUtilities::checkError;
+
+    // Create a local vector for just the monitor
+    Vec localFaceVec;
+    DMGetLocalVector(fluxDm, &localFaceVec) >> utilities::PetscUtilities::checkError;
+    VecZeroEntries(localFaceVec) >> utilities::PetscUtilities::checkError;
+
+    // Get the raw data for the global vectors
+    PetscScalar* localFaceArray;
+    VecGetArray(localFaceVec, &localFaceArray) >> utilities::PetscUtilities::checkError;
+
+    // Determine the size of data
+    PetscInt dataSize;
+    VecGetBlockSize(localFaceVec, &dataSize) >> utilities::PetscUtilities::checkError;
+
+    /** The TCP monitor must store one output for each of the radiation models that are in the vector of ray tracing solvers.
+     * The ratio of red to green intensities must be computed and output as well.
+     * It is not clear whether the red to green intensity ratio output should be implicitly defined in the input definition or whether there should be an explicit definition of which absorption model
+     * represents the red and green intensities respectively. THe definition of a helper class which represents two radiation solvers each carrying a red and green ray tracing solver would likely be
+     * beneficial for the definition of the models.
+     * First solve the radiation through each of the ray tracing solvers
+     */
+    for (auto& rayTracer : radiation) {
+        rayTracer->EvaluateGains(GetSolver()->GetSubDomain().GetSolutionVector(), GetSolver()->GetSubDomain().GetField("temperature"), GetSolver()->GetSubDomain().GetAuxVector());
+    }
+
+    /**
+     * After the radiation solution is computed, then the intensity of the individual radiation solutions can be output for each face.
+     */
+    if (locXArray && localFaceArray) {
+        // March over each cell in the face dm
+        PetscInt cStart, cEnd;
+        DMPlexGetHeightStratum(fluxDm, 0, &cStart, &cEnd) >> utilities::PetscUtilities::checkError;
+
+        // get the mapping information
+        IS faceIs;
+        const PetscInt* faceToBoundary = nullptr;
+        DMPlexGetSubpointIS(fluxDm, &faceIs) >> utilities::PetscUtilities::checkError;
+        ISGetIndices(faceIs, &faceToBoundary) >> utilities::PetscUtilities::checkError;
+
+        for (PetscInt c = cStart; c < cEnd; ++c) {
+            for (int i = 0; i < int(radiation.size()); i++) {
```

Comment:
I don't see any reason to cast to an int here, I would std::size_t instead of int.